### PR TITLE
Update outdated Limitations Page

### PR DESCRIPTION
**Pull request by javimbk**
Hello guys, first of all, thank you for the work that you are doing with the library. It is really amazing, been reading about it for the last day or two and I am planning on using it for my next project.

When checking the website I realized that you had a Limitations section, which I always find appealing, and I think it has been outdated for a while since you guys wrote it.

I took the liberty of updating it after checking the state of said issues. You can find the reasoning behind every removal on self-comments on the diff.

Hope this helps give a better picture of the things that may be missing!

### CODE REVIEW

File: packages/docs/src/pages/limitations.mdx [hunk 1]
```diff
@@ -30,11 +30,8 @@ If a dynamic value resolves to a static constant value however, it will work!
 Some behavior and features will be implemented in the future.
 Follow these issues to stay updated.
 
-- [Conditional CSS rules for styled](https://github.com/atlassian-labs/compiled/issues/390)
```
Comment (javimbk) [inline packages/docs/src/pages/limitations.mdx:33]: **note:** closed on https://github.com/atlassian-labs/compiled/pull/788

File: packages/docs/src/pages/limitations.mdx [hunk 2]
```diff
@@ -30,11 +30,8 @@ If a dynamic value resolves to a static constant value however, it will work!
...
 - [Conditional CSS rules for ClassNames](https://github.com/atlassian-labs/compiled/issues/391)
-- [CSS mixin api](https://github.com/atlassian-labs/compiled/issues/313)
```
Comment (javimbk) [inline packages/docs/src/pages/limitations.mdx:35]: **note:** closed on https://github.com/atlassian-labs/compiled/pull/696

File: packages/docs/src/pages/limitations.mdx [hunk 3]
```diff
@@ -30,11 +30,8 @@ If a dynamic value resolves to a static constant value however, it will work!
...
 - [Global component](https://github.com/atlassian-labs/compiled/issues/62)
-- [Animation keyframes](https://github.com/atlassian-labs/compiled/issues/14)
```
Comment (javimbk) [inline packages/docs/src/pages/limitations.mdx:37]: **note:** closed on https://github.com/atlassian-labs/compiled/pull/812

**Review by JakeLane — APPROVED** [review]
Thanks for your contribution!

**Comment by JakeLane** [comment]
We'll be going through the docs at some stage to update the content a bit with our recommendations. There are a few cases where we chose to compromise and not implement certain use cases.

**Pull request merged**